firmware/quadruna/BMS/src/app/app_accumulator.c

```diff
@@ -71,181 +58,301 @@ typedef enum
+        .pack_voltage     = 0.0f,
+    };
     // Find the min and max voltages
-    for (uint8_t segment = 0U; segment < ACCUMULATOR_NUM_SEGMENTS; segment++)
+    for (uint8_t segment = 0U; segment < NUM_SEGMENTS; segment++)
     {
-        for (uint8_t cell = 0U; cell < ACCUMULATOR_NUM_SERIES_CELLS_PER_SEGMENT; cell++)
+        for (uint8_t cell = 0U; cell < CELLS_PER_SEGMENT; cell++)
         {
             // Collect each cell voltage to find the min/max
-            const float cell_voltage = io_ltc6813CellVoltages_getCellVoltage(segment, cell);
-
+            const float cell_voltage = cell_voltages[segment][cell];
             // Get the minimum cell voltage
-            if (cell_voltage < temp_voltage_stats.min_voltage.voltage)
+            if (cell_voltage < voltage_stats.min_voltage_cell.voltage)
             {
-                temp_voltage_stats.min_voltage.voltage = cell_voltage;
-                temp_voltage_stats.min_voltage.segment = segment;
-                temp_voltage_stats.min_voltage.cell    = cell;
+                voltage_stats.min_voltage_cell.voltage = cell_voltage;
+                voltage_stats.min_voltage_cell.segment = segment;
+                voltage_stats.min_voltage_cell.cell    = cell;
             }
-
             // Get the maximum cell voltage
-            if (cell_voltage > temp_voltage_stats.max_voltage.voltage)
+            if (cell_voltage > voltage_stats.max_voltage_cell.voltage)
             {
-                temp_voltage_stats.max_voltage.voltage = cell_voltage;
-                temp_voltage_stats.max_voltage.segment = segment;
-                temp_voltage_stats.max_voltage.cell    = cell;
+                voltage_stats.max_voltage_cell.voltage = cell_voltage;
+                voltage_stats.max_voltage_cell.segment = segment;
+                voltage_stats.max_voltage_cell.cell    = cell;
             }
-
             // Sum the voltages into a segment voltage
-            temp_voltage_stats.segment_voltages[segment] += cell_voltage;
+            voltage_stats.segment_voltages[segment] += cell_voltage;
         }
-
         // Sum the segment voltages into a pack voltage
-        temp_voltage_stats.pack_voltage += temp_voltage_stats.segment_voltages[segment];
+        voltage_stats.pack_voltage += voltage_stats.segment_voltages[segment];
+    }
+}
+
+typedef struct
+{
+    uint8_t segment;
+    uint8_t cell;
+    float   temp;
+} CellTemp;
+typedef struct
+{
+    CellTemp min_temp_cell;
+    CellTemp max_temp_cell;
+    float    avg;
+} TempStats;
+static TempStats temp_stats;
+
+static void calculateTemperatureStats(void)
+{
+    uint8_t temp_fails = 0;
+    for (int i = 0; i < NUM_SEGMENTS; i++)
+    {
+        for (int j = 0; j < THERMISTOR_REGISTER_GROUPS; j++)
+        {
+            if (!temp_read_success[i][j])
+            {
+                temp_fails++;
+            }
+        }
     }
+    if (temp_fails > 0)
+    {
+        conceq_fails++;
```

Comment:
Is this “consecutive failures”? I’d revise the variable name here to be more clear